drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -326,6 +326,14 @@ int intel_adsp_hda_dma_stop(const struct device *dev, uint32_t channel)
 
 	intel_adsp_hda_disable(cfg->base, cfg->regblock_size, channel);
 
+	/* host dma needs some cycles to completely stop */
+	if (cfg->direction == HOST_TO_MEMORY || cfg->direction == MEMORY_TO_HOST) {
+		if (!WAIT_FOR(!(*DGCS(cfg->base, cfg->regblock_size, channel) & DGCS_GBUSY), 1000,
```

Comment:
Would be cleaner to abstract out the register math.  Also the timeout is in units of cycles, so you probably want to express this in terms of some real unit, e.g.:

```
     static bool hda_ready(struct device *dev) { ... }

     if (!(WAIT_FOR(hda_ready(dev), k_us_to_cyc_ceil32(10), k_busy_wait(1))) {
        return -EBUSY;
    }
```

Finally, note that k_busy_wait() is already spinning on the CPU needlessly.  Unless the hardware has trouble with banging on its registers, this will perform better with an empty statement like "`arch_nop()`", or I guess `arch_spin_relax()` if you want to be fancy.

Reply:
timeout is in us by documentation, if that's not the case then we need to fix either the macro or the docs

Reply:
Ah, you're right.  Misremembering on my part.  1000us is a fine fallback timeout.